kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManager.java

```diff
@@ -68,9 +70,8 @@ private static ExecutorService queryThreadPool() {
                     CompletableFuture.supplyAsync(
                             () -> chunk.query(query),
                             RequestContext.makeContextPropagating(queryExecutorService))
-                        // TODO: this will not cancel lucene query. Use ExitableDirectoryReader in
-                        // the future and pass this timeout
-                        .orTimeout(QUERY_TIMEOUT_SECONDS, TimeUnit.SECONDS))
+                        .orTimeout(
+                            LOCAL_QUERY_TIMEOUT_DURATION.toMillis() - 100, TimeUnit.MILLISECONDS))
```

Comment:
We should still keep the TODO since we don't cancel the underlying lucene query yet.

Also, we should also extract `LOCAL_QUERY_TIMEOUT_DURATION.toMillis() - 100` into a local variable and add a comment on why it's needed. Also, please document the magic number 100 that way we remember why we set it to that number later and what it can be updated to.